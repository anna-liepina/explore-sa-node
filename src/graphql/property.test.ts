//@ts-nocheck
describe('GraphQL: Property', () => {
    describe('propertySearch [presets: perPage 3]', () => {
        it(`by postcode wildcard`, async () => {
            const { data: { propertySearch: data } } = await query({
                query: `
                {
                    propertySearch(postcodePattern: "E20", perPage: 3) {
                        saon
                        paon
                        street
                        postcode {
                            postcode
                        }
                        transactions {
                            date
                        }
                    }
                }`
            });

            expect(data).toMatchSnapshot();
        });

        it(`by postcode wildcard: 2nd page`, async () => {
            const { data: { propertySearch: data } } = await query({
                query: `
                {
                    propertySearch(postcodePattern: "E20", perPage: 3, page: 2) {
                        saon
                        paon
                        street
                        postcode {
                            postcode
                        }
                        transactions {
                            date
                        }
                    }
                }`
            });

            expect(data).toMatchSnapshot();
        });
    });

    describe('propertySearchInRange [presets: perPage 2]', () => {
        it(`by coordinates, in kilometers`, async () => {
            const { data: { propertySearchInRange: data } } = await query({
                query: `
                {
                    propertySearchInRange(
                        pos: {
                            lat: 51.547915089,
                            lng: -0.00743124
                        },
                        range: 10,
                        perPage: 3
                    ) {
                        saon
                        paon
                        street
                        postcode {
                            postcode
                            lat
                            lng
                        }
                        transactions {
                            date
                        }
                    }
                }`
            });

            expect(data).toMatchSnapshot();
        });

        it(`by coordinates, in miles`, async () => {
            const { data: { propertySearchInRange: data } } = await query({
                query: `
                {
                    propertySearchInRange(
                        pos: {
                            lat: 51.547915089,
                            lng: -0.00743124
                        },
                        rangeUnit: ml,
                        range: 10,
                        perPage: 2
                    ) {
                        saon
                        paon
                        street
                        postcode {
                            postcode
                            lat
                            lng
                        }
                        transactions {
                            date
                        }
                    }
                }`
            });

            expect(data).toMatchSnapshot();
        });

        it(`by coordinates, in miles: 2nd page`, async () => {
            const { data: { propertySearchInRange: data } } = await query({
                query: `
                {
                    propertySearchInRange(
                        pos: {
                            lat: 51.547915089,
                            lng: -0.00743124
                        },
                        rangeUnit: ml,
                        range: 10,
                        perPage: 2,
                        page: 2
                    ) {
                        saon
                        paon
                        street
                        postcode {
                            postcode
                            lat
                            lng
                        }
                        transactions {
                            date
                        }
                    }
                }`
            });

            expect(data).toMatchSnapshot();
        });
    });
});
