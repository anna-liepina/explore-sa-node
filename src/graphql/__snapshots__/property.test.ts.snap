// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphQL: Property propertySearch [presets: perPage 3] by postcode wildcard 1`] = `
[
  {
    "address": "dd3c8e0d34ac3fd9dadc294e96ccd1cfa00e599a",
    "postcode": {
      "postcode": "E20 1AB",
    },
    "transactions": [
      {
        "date": "2019-04-25",
      },
    ],
  },
  {
    "address": "dea9c5dd93a7c61fa0bcebb9ea9f2814846203d1",
    "postcode": {
      "postcode": "E20 1AB",
    },
    "transactions": [
      {
        "date": "2019-04-26",
      },
    ],
  },
  {
    "address": "45c7a465a6da7fc33b9d4577cacafa53e849dda6",
    "postcode": {
      "postcode": "E20 1AB",
    },
    "transactions": [
      {
        "date": "2019-03-25",
      },
    ],
  },
]
`;

exports[`GraphQL: Property propertySearch [presets: perPage 3] by postcode wildcard: 2nd page 1`] = `
[
  {
    "address": "8b7de6d263024ec9713f5279e13aadee8d07f7c8",
    "postcode": {
      "postcode": "E20 1AB",
    },
    "transactions": [
      {
        "date": "2022-09-16",
      },
    ],
  },
  {
    "address": "77ef51b9253e7978202b37dc46113366bcddaa14",
    "postcode": {
      "postcode": "E20 1AB",
    },
    "transactions": [
      {
        "date": "2022-06-06",
      },
    ],
  },
  {
    "address": "abceb3e60dc1a7d71b9756bf1ddb58de20cd5002",
    "postcode": {
      "postcode": "E20 1AE",
    },
    "transactions": [
      {
        "date": "2017-11-10",
      },
    ],
  },
]
`;

exports[`GraphQL: Property propertySearchInRange [presets: perPage 2] by coordinates, in kilometers 1`] = `
[
  {
    "address": "dd3c8e0d34ac3fd9dadc294e96ccd1cfa00e599a",
    "postcode": {
      "lat": 51.548136,
      "lng": -0.007724,
      "postcode": "E20 1AB",
    },
    "transactions": [
      {
        "date": "2019-04-25",
      },
    ],
  },
  {
    "address": "dea9c5dd93a7c61fa0bcebb9ea9f2814846203d1",
    "postcode": {
      "lat": 51.548136,
      "lng": -0.007724,
      "postcode": "E20 1AB",
    },
    "transactions": [
      {
        "date": "2019-04-26",
      },
    ],
  },
  {
    "address": "45c7a465a6da7fc33b9d4577cacafa53e849dda6",
    "postcode": {
      "lat": 51.548136,
      "lng": -0.007724,
      "postcode": "E20 1AB",
    },
    "transactions": [
      {
        "date": "2019-03-25",
      },
    ],
  },
]
`;

exports[`GraphQL: Property propertySearchInRange [presets: perPage 2] by coordinates, in miles 1`] = `
[
  {
    "address": "dd3c8e0d34ac3fd9dadc294e96ccd1cfa00e599a",
    "postcode": {
      "lat": 51.548136,
      "lng": -0.007724,
      "postcode": "E20 1AB",
    },
    "transactions": [
      {
        "date": "2019-04-25",
      },
    ],
  },
  {
    "address": "dea9c5dd93a7c61fa0bcebb9ea9f2814846203d1",
    "postcode": {
      "lat": 51.548136,
      "lng": -0.007724,
      "postcode": "E20 1AB",
    },
    "transactions": [
      {
        "date": "2019-04-26",
      },
    ],
  },
]
`;

exports[`GraphQL: Property propertySearchInRange [presets: perPage 2] by coordinates, in miles: 2nd page 1`] = `
[
  {
    "address": "45c7a465a6da7fc33b9d4577cacafa53e849dda6",
    "postcode": {
      "lat": 51.548136,
      "lng": -0.007724,
      "postcode": "E20 1AB",
    },
    "transactions": [
      {
        "date": "2019-03-25",
      },
    ],
  },
  {
    "address": "8b7de6d263024ec9713f5279e13aadee8d07f7c8",
    "postcode": {
      "lat": 51.548136,
      "lng": -0.007724,
      "postcode": "E20 1AB",
    },
    "transactions": [
      {
        "date": "2022-09-16",
      },
    ],
  },
]
`;
